### ISSUE TITLE
Support for embedded data with useScript

### ISSUE THREAD

**Issue by sasivarnan** [JoviDeCroock/hoofd#43]
I am trying to migrate to `hoofd` library on my [blog](geekscreed.com) which currently uses `react-helmet`. 

I almost migrated everything except Rich Text Data which needs to be embedded within the script tag.

Is it possible to add support for embedding data with `useScript`?

References that might be useful
1. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/script#embedding_data_in_html 
2. https://developers.google.com/search/docs/guides/intro-structured-data

**Comment by JoviDeCroock** [comment]
You can probably achieve this in a similar way to embedding a script into `useScript` by means of `data:application/javascript,alert("yolo")` but then in your case `data:application/ld+json,JSON.stringify(yourjson)`

**Comment by sasivarnan** [comment]
Thanks for ultra fast response. ⚡️

I have tried that before raising the issue. But the browser was throwing `DOMException: Document.querySelectorAll:...is not a valid selector` error.

I even tried doing the following

```js
    src: "data:application/json," + encodeURI(organizationLdJson),
```

But the [Rich Results Test Tool](https://search.google.com/test/rich-results) isn't unable to extract out the information.

**Comment by JoviDeCroock** [comment]
Ah, makes sense 😅 we lack a way to uniquely identify this 🤔 we could allow folks to specify an `id` and we use that when we are able to

**Comment by sasivarnan** [comment]
Thanks for the input. I have implemented a quick solution with your approach. 

Kindly have a look at it once. https://github.com/sasivarnan/hoofd/commit/fd6613067624c2c58be9579f043804d05a3521d0 

If you are happy with the solution, I will work on some more test cases and raise a PR.

**Comment by JoviDeCroock** [comment]
That looks good to me yes 😅 thanks!

**Comment by JoviDeCroock** [comment]
Published as 1.3.0
